spotinst_sdk2/models/ocean/aws/__init__.py

```diff
@@ -1,370 +1,918 @@
+        self.end_time = end_time
+        self.aggregated_filter = aggregated_filter
+        self.group_by = group_by
+        self.start_time = start_time
 
-	def toJSON(self):
-		return json.dumps(self, default=lambda o: o.__dict__,
+
+class AggregatedClusterCostRequest:
+    def __init__(self, aggregated_cluster_costs: AggregatedClusterCosts = none):
+        self.end_time = aggregated_cluster_costs.end_time
+        self.start_time = aggregated_cluster_costs.start_time
+        self.group_by = aggregated_cluster_costs.group_by
+        self.filter = aggregated_cluster_costs.aggregated_filter
+
+    def toJSON(self):
+        return json.dumps(self, default=lambda o: o.__dict__,
+                          sort_keys=True, indent=4)
+
+
+class Roll:
+    """
+    # Arguments
+    batch_min_healthy_percentage: int
+    batch_size_percentage: int
+    comment: str
+    disable_launch_spec_auto_scaling: bool
+    instance_ids: List[str]
+    launch_spec_ids: List[str]
+    respect_pdb: bool
+    """
+    def __init__(
+            self,
+            batch_min_healthy_percentage: int = none,
+            batch_size_percentage: int = none,
+            comment: str = none,
+            disable_launch_spec_auto_scaling: bool = none,
+            launch_spec_ids: List[str] = none,
+            instance_ids: List[str] = none,
+            respect_pdb: bool = none):
+        self.batch_min_healthy_percentage = batch_min_healthy_percentage
+        self.batch_size_percentage = batch_size_percentage
+        self.comment = comment
+        self.disable_launch_spec_auto_scaling = disable_launch_spec_auto_scaling
+        self.instance_ids = instance_ids
+        self.launch_spec_ids = launch_spec_ids
+        self.respect_pdb = respect_pdb
+
+
+class ClusterRollInitiateRequest:
+    def __init__(self, roll: Roll = none):
+        self.roll = roll
+
+    def toJSON(self):
+        return json.dumps(self, default=lambda o: o.__dict__,
+                          sort_keys=True, indent=4)
+
+
+class Status:
+    """
+    # Arguments
+    status: str
+    """
+    def __init__(
+            self,
+            status: str = none):
+        self.status = status
+
+
+class UpdateRoll:
+    """
+    # Arguments
+    roll: Status
+    """
+    def __init__(
+            self,
+            roll: Status = none):
+        self.roll = roll
+
+
+class ClusterRollUpdateRequest:
+    def __init__(self, updateRoll: UpdateRoll = none):
+        self.roll = updateRoll.roll
+
+    def toJSON(self):
+        return json.dumps(self, default=lambda o: o.__dict__,
+                          sort_keys=True, indent=4)
+
+
+class InstanceTypesFilterSimulationRequest:
+    def __init__(self, instance_type_filter: InstanceTypesFilters = none):
+        self.filter = instance_type_filter
+
+    def toJSON(self):
+        return json.dumps(self, default=lambda o: o.__dict__,
                           sort_keys=True, indent=4)
 
+
+class LaunchNodes:
```

Comment:
Do we really need this class? The API can be directly passed the "amount" value, I believe.

Reply:
DEBUG:spotinst_sdk2.client:400
DEBUG:spotinst_sdk2.client:{"status": {"code": 400, "message": "Bad Request"}, "errors": [{"code": "ValidationError", "message": "\"launchRequest\" is required", "field": "body:launchRequest"}, {"code": "ValidationError", "message": "\"amount\" is not allowed", "field": "body:amount"}]}


This class is required